sympy/combinatorics/tensor_can.py

```diff
@@ -1141,11 +1141,11 @@ def _get_bsgs(G, base, gens, free_indices):
         a.extend(ind2)
         a.extend(ind1)
         base_comm.append(ind1[0])
-        a.extend(list(range(ind2[-1] + 1, nr)))
+        a.extend(range(ind2[-1] + 1, nr))
         if sym == 0:
-            a.extend([nr, nr + 1])
+            a.extend((nr, nr + 1))
```

Comment:
Is this sort of thing actually faster?

I get that if the tuple elements are compile-time constants then the tuple can be pre-created and marshalled into the pyc file but if the tuple elements depend on local variables as is the case here then we're just talking about the difference between the time to construct a tuple vs a list:
```python
def f1(l, a):
    for _ in range(100):
        l.extend([a, a+1])

def f2(l, a):
    for _ in range(100):
        l.extend((a, a+1))

def f3(l, a):
    for _ in range(100):
        l += [a, a+1]

def f4(l, a):
    for _ in range(100):
        l.append(a)
        l.append(a+1)
```
Timings:
```python
In [28]: l = []

In [29]: %timeit f1(l, 5)
30 µs ± 105 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [30]: l = []

In [31]: %timeit f2(l, 5)
26.3 µs ± 107 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [32]: l = []

In [33]: %timeit f3(l, 5)
23.1 µs ± 2.55 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [34]: l = []

In [35]: %timeit f4(l, 5)
31.5 µs ± 244 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```
Looks like `+=` with a list is fastest.

Reply:
I didn't consider adding to the list as an option, but it indeed looks faster. I read https://stackoverflow.com/questions/14446128/python-append-vs-extend-efficiency where no one seem to have thought about it either.

Reply:
I guess there is a threshold for when `+=` is faster? Either way, for most cases I kept the tuples and extended.

Reply:
I think that `+=` with a list is faster when the list of new items is small (e.g. length 2 here). The reason is just that in CPython an operator like `+` is more efficient than a function/method call for e.g. `extend`.